cfme/infrastructure/virtual_machines.py

```diff
@@ -997,17 +1099,19 @@ def am_i_here(self, *args, **kwargs):
             provider = self.obj.provider.name
         else:
             raise DestinationNotFound("the destination isn't found")
-        return match_page(summary='VM or Templates under Provider "{}"'.format(provider))
+        return (self.view.sidebar.vmstemplates.is_opened and self.view.entities.title.text ==
+                'VM or Templates under Provider "{}"'.format(provider))
 
 
 @navigator.register(Template, 'AllForDatacenter')
 @navigator.register(Vm, 'AllForDatacenter')
 class VmAllWithTemplatesForDatacenter(CFMENavigateStep):
+    VIEW = VmsTemplatesAllView
```

Comment:
Same as Provider view above, I think this needs to be a separate view with `is_displayed` containing the checks for the title and accordion selection.